### DISCUSSION TITLE
Token based auth for Clickhouse using auth.token

### DISCUSSION THREAD

**Discussion by DSdatsme** [vectordotdev/vector#18105] [Q&A] [answered]
Hi,
I was trying to figure out how to get auth token as mentioned here in docs.
https://vector.dev/docs/reference/configuration/sinks/clickhouse/#auth.token

The Clickhouse docs do not mention anything about this method but if Vector supports it, I'd like to know how to set it up.
Currently, I am using a user-password-based strategy, but auth token-based method looks like a neater way of pushing data.

for more context, we have a simple clickhouseDB server and we are using clickhouse sink to push data.

Any help on this is appreciated, thanks.

**Comment by jszwedko** [comment, marked as answer]
Hi @DSdatsme !

I'm also not seeing any support in Clickhouse for token-based authentication. I think that may have just been included in the Vector `clickhouse` sink as it reuses the HTTP authentication mechanism for other HTTP-based sinks. All I can see supported for Clickhouse is username/password authentication.
